### PR TITLE
More string fixes

### PR DESCRIPTION
**Pull request by jgallagher59701**
Yup. more still more.

### CODE REVIEW

File: modules/w10n_handler/w10n_utils.cc [hunk 1]
```diff
@@ -113,17 +113,27 @@ void eval_resource_path(const string &w10nResourceId, const string &catalogRoot,
     remainder = rem;
 
     // Remove leading slash
-    if (rem[0] == '/') rem = rem.substr(1, rem.length() - 1);
-
-    // Remove trailing slash
-    if (rem[rem.length() - 1] == '/') rem = rem.substr(0, rem.length() - 1);
+    if (rem.length() > 0 && rem[0] == '/') rem = rem.substr(1, rem.length() - 1);
```
Comment (ndp-opendap) [inline modules/w10n_handler/w10n_utils.cc:116]: ibid

File: modules/w10n_handler/w10n_utils.cc [hunk 2]
```diff
@@ -113,17 +113,27 @@ void eval_resource_path(const string &w10nResourceId, const string &catalogRoot,
...
+
+    // Remove trailing slash.
+    BESUtil::trim_if_trailing_slash(rem);
+#if 0
+    if (rem.length() > 0 && rem[rem.length() - 1] == '/')
```
Comment (ndp-opendap) [inline modules/w10n_handler/w10n_utils.cc:121]: ibid

File: modules/w10n_handler/w10n_utils.cc [hunk 3]
```diff
@@ -113,17 +113,27 @@ void eval_resource_path(const string &w10nResourceId, const string &catalogRoot,
...
 
     // full path of the thing to check
     string fullpath = catalogRoot;
     // Remove leading slash
-    if (fullpath[fullpath.length() - 1] == '/') {
-        fullpath = fullpath.substr(0, fullpath.length() - 1);
+    BESUtil::trim_if_trailing_slash(fullpath);
+
+#if 0
+    if (fullpath.length() > 0 && fullpath[fullpath.length() - 1] == '/') {
```
Comment (ndp-opendap) [inline modules/w10n_handler/w10n_utils.cc:132]: ibid

**Review by ndp-opendap — CHANGES_REQUESTED** [review]
Since there is all this clean up, and since the new methods use 
```
!foo.empty()
```
instead of
```
foo.length() > 0
```
I think the other uses of `foo.length() > 0` should be replaced.

File: http/HttpUtils.cc [hunk 4]
```diff
@@ -133,13 +134,17 @@ void get_type_from_disposition(const string &disp, string &type)
...
 
+            BESUtil::trim_if_surrounding_quotes(filename);
+
+#if 0
             // now see if it's wrapped in quotes
             if (filename[0] == '"') {
                 filename = filename.substr(1);
             }
-            if (filename[filename.length() - 1] == '"') {
+            if (filename.length() > 0 && filename[filename.length() - 1] == '"') {
```
[the remark this answers is not in the document]
Comment (jgallagher59701) [reply http/HttpUtils.cc:144]: Odd. The branch for this PR should have had

            BESUtil::trim_if_surrounding_quotes(filename);
#if 0
            // now see if it's wrapped in quotes
            if (filename[0] == '"') {
                filename = filename.substr(1);
            }
            if (filename.length() > 0 && filename[filename.length() - 1] == '"') {
                filename = filename.substr(0, filename.length() - 1);
            }
#endif

for that region. I'll double check that the PR actually _has_ the changes I made.

File: modules/cmr_module/unused/CmrUtils.cc [hunk 5]
```diff
@@ -237,13 +237,16 @@ void CmrUtils::Get_type_from_disposition(const string &disp, string &type)
...
             }
 
+            BESUtil::trim_if_surrounding_quotes(filename);
+#if 0
             // now see if it's wrapped in quotes
             if (filename[0] == '"') {
                 filename = filename.substr(1);
             }
-            if (filename[filename.length() - 1] == '"') {
+            if (filename.length() > 0 && filename[filename.length() - 1] == '"') {
```
[the remark this answers is not in the document]
Comment (jgallagher59701) [reply modules/cmr_module/unused/CmrUtils.cc:246]: This code is unused

File: modules/fileout_netcdf/FONcRequestHandler.cc [hunk 6]
```diff
@@ -106,7 +106,11 @@ static void read_key_value(const string &key_name, string &key, const string &de
     TheBESKeys::TheKeys()->get_value(key_name, key, key_found);
     // 'key' holds the string value at this point if key_found is true
     if (key_found) {
-        if (key[key.length() - 1] == '/') key.erase(key.length() - 1);
+        BESUtil::trim_if_trailing_slash(key);
+#if 0
+        if (key.length() > 0 && key[key.length() - 1] == '/')
```
[the remark this answers is not in the document]
Comment (jgallagher59701) [reply modules/fileout_netcdf/FONcRequestHandler.cc:111]: #if 0

File: modules/w10n_handler/w10n_utils.cc (hunk 1 above)
Comment (jgallagher59701) [reply modules/w10n_handler/w10n_utils.cc:116]: fixed

File: modules/w10n_handler/w10n_utils.cc (hunk 2 above)
Comment (jgallagher59701) [reply modules/w10n_handler/w10n_utils.cc:121]: #if 0

File: modules/w10n_handler/w10n_utils.cc (hunk 3 above)
Comment (jgallagher59701) [reply modules/w10n_handler/w10n_utils.cc:132]: #if 0

**Comment by jgallagher59701** [comment]
I'll remove the #if 0 blocks and resubmit to make the changes cleaner. Also, I'll look for more places in the code where we do this.

**Comment by jgallagher59701** [comment]
Various changes made. More places fixed. More tests added.

**Pull request merged**